src/redis-cli.c

```diff
@@ -2425,13 +2425,15 @@ static int confirmWithYes(char *msg, int ignore_force) {
 }
 
 static int issueCommandRepeat(int argc, char **argv, long repeat) {
-    while (1) {
-        if (!config.eval_ldb && /* In debugging mode, let's pass "help" to Redis. */
-            (!strcasecmp(argv[0],"help") || !strcasecmp(argv[0],"?"))) {
-            cliOutputHelp(--argc, ++argv);
-            return REDIS_OK;
-        }
+    /* In debugging mode, let's pass "help" to Redis.
```

Comment:
i don't understand the debugging mode reference, but shouldn't it be negated?

Reply:
I haven't used lua debugger much. but look like there is a issue, i am taking a look..

cli (6.2) with 6.2 server
```
[root@binblog redis]# src/redis-cli --version
redis-cli 6.2.0 (git:445aa844)
[root@binblog redis]# src/redis-cli --ldb --eval test.lua test 1
Lua debugging session started, please use:
quit    -- End the session.
restart -- Restart the script in debug mode again.
help    -- Show Lua script debugging commands.

* Stopped at 1, stop reason = step over
-> 1   local num = redis.call('GET', KEYS[1]);  
lua debugger>
```

redis-cli (unstable) with 6.2 server (or with unstable server)
```
[root@binblog redis]# src/redis-cli --version
redis-cli 255.255.255 (git:311a757c)
[root@binblog redis]# src/redis-cli --ldb --eval test.lua test 1
Lua debugging session started, please use:
quit    -- End the session.
restart -- Restart the script in debug mode again.
help    -- Show Lua script debugging commands.

* Stopped at 1, stop reason = step over
-> 1   local num = redis.call('GET', KEYS[1]);  
redis-cli: redis-cli.c:718: cliCountCommands: Assertion `commandTable->element[i]->type == 1' failed.
Aborted
```

Reply:
it look like `COMMAND DOCS` or `COMMAND` in ldb mode, we will return an array, only with one element
and the element type is `REDIS_REPLY_STATUS`, and the result is `<error> Unknown Redis Lua debugger command or wrong number of arguments.`

and in 6.2 (or in the very old way), happened to be fixed by it #3680  (if we remove it in 6.2.0, also will crash the cli)
```
entry->type != REDIS_REPLY_ARRAY
```

Reply:
ok,  so the comment isn't negated.
i.e. in LDB mode, we want to pass the "help" to redis rather than handle it by the CLI.
maybe the comment can be slightly improved.

anyway, what's the verdict with your other finding above? is another change needed?

Reply:
yes, the another change is needed, currently redis-cli --ldb --evel is broken
the example i show above, not because of my changes, was broken in #10043

Reply:
ok, please fix here. additional adjustments for the same original change.

Reply:
hope you can understand my top comment (It looks a little verbose)